fieldservice/views/fsm_order.xml

```diff
@@ -0,0 +1,281 @@
+            <form string="Orders">
+                <header>
+                    <button id="action_confirm"
+                            name="action_confirm" string="Confirm"
+                            class="oe_highlight"
+                            type="object" groups="fieldservice.group_fsm_dispatcher"
+                            attrs="{'invisible': [('stage_id', '!=', %(fieldservice.fsm_stage_new)d)]}"/>
+                    <button id="action_schedule"
+                            name="action_schedule" string="Schedule"
+                            class="oe_highlight"
+                            type="object" groups="fieldservice.group_fsm_dispatcher"
+                            attrs="{'invisible': [('stage_id', '!=', %(fieldservice.fsm_stage_confirmed)d)]}"/>
+                    <button id="action_assign"
+                            name="action_assign" string="Assign"
+                            class="oe_highlight"
+                            type="object" groups="fieldservice.group_fsm_dispatcher"
+                            attrs="{'invisible': [('stage_id', '!=', %(fieldservice.fsm_stage_scheduled)d)]}"/>
+                    <button id="action_plan"
+                            name="action_plan" string="Plan"
+                            class="oe_highlight"
+                            type="object" groups="fieldservice.group_fsm_dispatcher"
+                            attrs="{'invisible': [('stage_id', '!=', %(fieldservice.fsm_stage_assigned)d)]}"/>
+                    <button id="action_enroute"
+                            name="action_enroute" string="En Route"
+                            class="oe_highlight"
+                            type="object" groups="fieldservice.group_fsm_dispatcher"
+                            attrs="{'invisible': [('stage_id', '!=', %(fieldservice.fsm_stage_planned)d)]}"/>
+                    <button id="action_start"
+                            name="action_start" string="Start"
+                            class="oe_highlight"
+                            type="object" groups="fieldservice.group_fsm_dispatcher"
+                            attrs="{'invisible': [('stage_id', '!=', %(fieldservice.fsm_stage_enroute)d)]}"/>
+                    <button id="action_complete"
+                            name="action_complete" string="Complete"
+                            class="oe_highlight"
+                            type="object" groups="fieldservice.group_fsm_dispatcher"
+                            attrs="{'invisible': [('stage_id', '!=', %(fieldservice.fsm_stage_started)d)]}"/>
+                    <button id="action_cancel"
+                            name="action_cancel" string="Cancel"
+                            type="object" groups="fieldservice.group_fsm_dispatcher"
+                            attrs="{'invisible': [('stage_id', '=', %(fieldservice.fsm_stage_completed)d)]}"/>
+                    <field name="stage_id" widget="statusbar"/>
```

Comment:
This is a hard coded stage list and workflow.
I would prefer a more flexible stage/state combination, more friendly for each customer's specific workflow.
We can have a separate discussion on this topic if you like.

Reply:
Thanks